v3/app/models/user.rb

```diff
@@ -0,0 +1,114 @@
+class User < ActiveRecord::Base
+
+    has_secure_password
+
+    def self.validate_token(options, check_role_id = 0) # check_role_id default to 0 for 'any'
+      option_token = options[:token].to_s
+      option_ip = options[:ip].to_s
+      option_timenow = Time.now.utc
+      timeok = (option_timenow - 15.minutes).to_s[0,19]
```

Comment:
I had thought about making it an environment variable.  I will do that.

Note we could make it more easily configurable, but I do not think it is worth it.  Remember that the token gets refreshed on every (valid) API call to the DB, so as long as the user does *something* that triggers a valid token (and almost everything on the site will do that), then they will never see this.  Also, if they do see this then the front-end should handle it cleanly and seamlessly (something like a popup that says "your session has expired, please re-enter your password to continue").  I believe it would really only timeout if the user steps away from their system and comes back <n> minutes later...

Reply:
I'm just reading and reacting to the code that you have written. All of that other stuff in your note is not here.

Reply:
DONE.  Added SESSION_TIMEOUT env variable.